scripts/e2e/compare_metrics.py

```diff
@@ -0,0 +1,115 @@
+# Copyright (c) 2024 The Jaeger Authors.
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+import re
+import sys
+from pathlib import Path
+
+def read_metric_file(file_path):
+    with open(file_path, 'r') as f:
+        return f.readlines()
+
+def parse_metric_line(line):
+
+    # Skip empty lines or comments
+    line = line.strip()
+    if not line or line.startswith('#'):
+        return None
+    
+    # Extract metric name and labels section
+    match = re.match(r'([^{]+)(?:{(.+)})?(?:\s+[-+]?[0-9.eE+-]+)?$', line)
```

Comment:
it would be simpler if we use Prometheus python client to parse

https://github.com/prometheus/client_python/blob/5926a7c0ddb8755a5e504e6f077e081f5f4c7674/prometheus_client/parser.py#L13

You can install it just before running the script